### PR TITLE
licensure 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by marques-work**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by marques-work** [comment]
Tests fail `audit` because:

1. not "notable" enough
2. patches (explained in code comment)

Are these deal breakers?

File: Formula/licensure.rb [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
+  end
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/licensure --version")
```
Comment (carlocab) [inline Formula/licensure.rb:30]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
> Tests fail `audit` because:
> 
> 1. not "notable" enough
> 2. patches (explained in code comment)
> 
> Are these deal breakers?

Yes, sorry.

If you wish, you can host this formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

If you submit this again in the future, please provide a better test and use homebrew-core style when writing your commit message. See the style guide linked in the PR template for details.

Additionally, please run `brew audit --new licensure` as requested in the PR template so that we don't need to run CI before the errors are discovered.

**Comment by marques-work** [comment]
Hi there — thanks for the reply. Actually I did run the `brew audit —new
—strict —formula Formula/licensure.rb` locally, and it did not fail.

Is it supposed to? It seems like the CI tests do a more rigorous audit than
the command line.

**Comment by SMillerDev** [comment]
That should be `brew audit --new --strict --formula licensure`, does it fail when you do that?

**Comment by carlocab** [comment]
> where `<formula>` is the name of the formula you're submitting

This means you need to run
```
brew audit --new licensure
```
and not
```
brew audit --new path/to/licensure.rb
```
It's a common mistake that `brew` needs to handle better. But, in the meantime, please use the name of the formula and not a path to the formula file when running `brew audit`.

More generally: don't pass filenames/paths to `brew`. (again, something `brew` needs to handle better, but fixes are made on volunteer time so rough edges tend to stick around for longer than ideal.)

**Comment by marques-work** [comment]
Understood @SMillerDev @carlocab  -- though, still, it does not fail :).

```bash
$ brew audit --new --strict --formula licensure
$ echo $?
0
```

So, I think invoking `audit` like this is still different from what the tests do (and I haven't looked into how that works yet either).

FWIW, while I was authoring `licensure.rb`, `brew audit --new path/to/licensure.rb` was still giving me valuable feedback about errors I needed to correct, so it seemed to work even with the path argument (despite that it's not the recommended method).

**Comment by SMillerDev** [comment]
When I check out your branch I get:
```
/o/h/L/T/h/homebrew-core (add-licensure-package|✔) $ brew audit --new --strict --formula licensure
licensure:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
  * Formulae should not require patches to build. Patches should be submitted and accepted upstream first.
Error: 2 problems in 1 formula detected
```

**Comment by marques-work** [comment]
Even tried:

```bash
$ brew audit --new --strict --formula --online licensure
$ echo $?
0
```

And no luck.

**Comment by marques-work** [comment]
@SMillerDev interesting! I could not reproduce this locally. Are there some configurations I need to check with `brew`? Because that's the behavior I _should_ be seeing.

**Comment by marques-work** [comment]
Not sure if this is helpful, but I ran `brew doctor`:

```bash
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  licensure

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/marques/.asdf/shims/python3.7-config
  /Users/marques/.asdf/shims/python3.10-config
  /Users/marques/.asdf/shims/python3.7m-config
  /Users/marques/.asdf/shims/python-config
  /Users/marques/.asdf/shims/python3-config

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@3.10
```

And:

```bash
$ brew --version
Homebrew 3.6.2-26-g4441452
Homebrew/homebrew-core (git revision 6caacbe4e30; last commit 2022-09-26)
Homebrew/homebrew-cask (git revision 0970b180eba; last commit 2022-09-25)
```

I think the shell _shouldn't_ matter, but in case it does, I'm running `bash` (your prompt looks like `zsh`)

**Comment by SMillerDev** [comment]
You don't seem to be on this branch, so there is not formula to audit.

**Comment by marques-work** [comment]
I'm pretty sure I'm on my branch in my fork:

```bash
$ git rev-parse HEAD
80c05fb1f8075525d435f5805fe092b58a58152f
$ git rev-parse add-licensure-package
80c05fb1f8075525d435f5805fe092b58a58152f
```

The SHAs match -- unless I need to be on a different branch?

**Comment by SMillerDev** [comment]
That's not the hash `brew --version` reported when you posted it above though.

**Comment by marques-work** [comment]
Oh, I think I understand now. Makes sense and dunno why I didn't connect the dots. I need to go to `/usr/local/Homebrew` and switch branches to the PR ref, and not simply be in the working directory of my fork of `homebrew-core`.

Sorry for annoying you with the noob questions :).

**Comment by SMillerDev** [comment]
I'm always happy to help people trying to contribute to Homebrew.

**Pull request closed without merging**